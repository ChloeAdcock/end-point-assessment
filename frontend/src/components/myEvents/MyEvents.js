import React from "react";
import Drawer from "@material-ui/core/Drawer";
import Typography from "@material-ui/core/Typography";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Divider from "@material-ui/core/Divider";
import Button from "@material-ui/core/Button";
import formatDateTime from "../../helpers/formatDateTime";
import { useStyles } from "../../styles/drawerStyles";

function MyEvents(props) {
  const classes = useStyles();

  return (
    <Drawer variant="permanent">
      {!props.events ? (
        <Typography>You don't have any events</Typography>
      ) : (
        <List style={{ marginTop: "20%" }}>
          {props.events.map((event) => {
            return (
              <div id={event.id}>
                <ListItem className={classes.listItem}>
                  <ListItemText
                    primary={<Typography variant="h6">{event.name}</Typography>}
                    secondary={
                      <Typography>{formatDateTime(event.date_time)}</Typography>
                    }
                  />
                  <Button
                    onClick={() => props.handleClick(event)}
                    fullWidth
                    color="primary"
                    variant="contained"
                  >
                    Details
                  </Button>
                </ListItem>
                <Divider />
              </div>
            );
          })}
        </List>
      )}
    </Drawer>
  );
}

export default MyEvents;
