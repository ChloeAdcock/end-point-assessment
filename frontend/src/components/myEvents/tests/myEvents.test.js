import renderer from "react-test-renderer";
import MyEvents from "../MyEvents";

describe("The my events component", () => {
    it("should render events when they are passed down as props", () => {
      const tree = renderer
        .create(
            <MyEvents events={[{date_time: 'test'}]}/>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render no events found when there are no events", () => {
        const tree = renderer
          .create(
              <MyEvents/>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
  
  });