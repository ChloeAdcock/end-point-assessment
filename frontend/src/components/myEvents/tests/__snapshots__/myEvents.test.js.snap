// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The my events component should render events when they are passed down as props 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
  >
    <ul
      className="MuiList-root MuiList-padding"
      style={
        Object {
          "marginTop": "20%",
        }
      }
    >
      <div>
        <li
          className="MuiListItem-root makeStyles-listItem-1 MuiListItem-gutters"
          disabled={false}
        >
          <div
            className="MuiListItemText-root MuiListItemText-multiline"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
            />
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              NaN/NaN/NaN NaN:NaN
            </p>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Details
            </span>
          </button>
        </li>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </ul>
  </div>
</div>
`;

exports[`The my events component should render no events found when there are no events 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      You don't have any events
    </p>
  </div>
</div>
`;
